src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRazorViewEngineFileProviderAccessor.cs

```diff
@@ -18,7 +19,15 @@ public class DefaultRazorViewEngineFileProviderAccessor : IRazorViewEngineFilePr
         public DefaultRazorViewEngineFileProviderAccessor(IOptions<RazorViewEngineOptions> optionsAccessor)
         {
             var fileProviders = optionsAccessor.Value.FileProviders;
-            if (fileProviders.Count == 1)
+            if (fileProviders.Count == 0)
+            {
+                var message = Resources.FormatFileProvidersAreRequired(
+                    typeof(RazorViewEngineOptions).FullName,
+                    nameof(RazorViewEngineOptions.FileProviders),
+                    typeof(IFileProvider).FullName);
+                throw new ArgumentException(message, nameof(optionsAccessor));
```

Comment:
I don't understand why this is an error. If the list of file providers is empty you'll still have a reasonable and well defined experience